client/src/modules/cohort/components/CohortName.jsx

```diff
@@ -50,11 +55,25 @@ class CohortName extends PureComponent {
     const {
       details: { name: previousName }
     } = previousProps;
-    const hasNameBeenChanged = name !== previousName;
+    const { nameInputValue } = this.state;
+    const nameHasChanged = name !== previousName;
+    const dataHasChanged = previousName !== nameInputValue;
```

Comment:
inputValueHasChanged?

Reply:
I think data has changed is understandable enough.